### PR TITLE
openssl: bump to 1.0.2r release

### PR DESCRIPTION
**Pull request by jgehrcke**
## Corresponding DC/OS tickets (obligatory)

https://jira.mesosphere.com/browse/DCOS_OSS-4868 

## Checklist for all PRs

  - [x] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -13,7 +13,20 @@ Format of the entries must be.
 * Entry two with no-newlines. (DCOS_OSS_JIRA_2)
 ```
 
-## DC/OS 1.12.3 (Next Release. Please Update News Entries in this Section).
+## DC/OS 1.12.4 (Next release, please update entries in this section)
+
+### Notable changes
```
Comment (jgehrcke) [inline CHANGES.md]: Hey @orsenthil @catsouthard I thought we do this kind of change to CHANGES.md as part of "cutting a release" process. Will this be added to some TODO list that is to be checked of every release we cut?

Comment (jgehrcke) [reply CHANGES.md]: Related: https://github.com/dcos/dcos/pull/3486#issuecomment-424670155

**Comment by jgehrcke** [comment]
@mesosphere-mergebot label Ready For Review

File: packages/openssl/buildinfo.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "single_source" : {
     "kind": "url_extract",
-    "url": "https://openssl.org/source/openssl-1.0.2p.tar.gz",
-    "sha1": "f34b5322e92415755c7d58bf5d0d5cf37666382c"
+    "url": "https://www.openssl.org/source/openssl-1.0.2r.tar.gz",
+    "sha1": "b9aec1fa5cedcfa433aed37c8fe06b0ab0ce748d"
```
Comment (jongiddy) [inline packages/openssl/buildinfo.json]: Confirmed that this SHA matches the one in https://www.openssl.org/source/openssl-1.0.2r.tar.gz.sha1

**Pull request merged**